### PR TITLE
GEOMESA-1551 Fail writing features with invalid geometries or dates

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

**Comment by elahrvivaz** [comment]
Note: users will need to pre-validate geometries or wrap calls to `featureWriter.write` in a try/catch.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/XZ3WritableIndex.scala [hunk 1]
```diff
@@ -69,7 +69,11 @@ trait XZ3WritableIndex extends AccumuloWritableIndex {
                        (wf: AccumuloFeature): Array[Byte] = {
     val numSplits = splitArray.length
     val split = splitArray(wf.idHash % numSplits)
-    val envelope = wf.feature.getDefaultGeometry.asInstanceOf[Geometry].getEnvelopeInternal
+    val geom = wf.feature.getDefaultGeometry.asInstanceOf[Geometry]
+    if (geom == null) {
+      throw new IllegalArgumentException(s"Null geometry in feature ${wf.feature.getID}")
```
Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/XZ3WritableIndex.scala]: What does this do to bulk ingests with bad data?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/XZ3WritableIndex.scala]: our converter ingest will catch each failed sf and keep going - third party ingests would need to do the same...

Comment (jnh5y) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/z3/XZ3WritableIndex.scala]: Well, third parties can do as they like.  After you rebase, I'd say we can shippit!

**Comment by elahrvivaz** [comment]
rebased

**Pull request merged**